Tests/NIOTests/EventLoopFutureTest.swift

```diff
@@ -82,6 +82,59 @@ class EventLoopFutureTest : XCTestCase {
             XCTFail("error of wrong type \(e)")
         }
     }
+    
+    func testReduceWithAllSuccesses() throws {
+        let eventLoop = EmbeddedEventLoop()
+        let promises: [EventLoopPromise<Int>] = (0..<5).map { (_: Int) in eventLoop.newPromise() }
+        let futures = promises.map { $0.futureResult }
+        
+        let fN: EventLoopFuture<[Int]> = EventLoopFuture<[Int]>.reduce([], futures, eventLoop: eventLoop, reducer: {$0 + [$1]})
```

Comment:
This `+` does look like quadratic operation. Does it make sense to create some blueprint or comment or a specialized function for such kind of array-based accumulator that makes assembling the result linear?

Reply:
@vlm That is a perfect use-case for `reduce(into:)`, that I mentioned [here](https://github.com/apple/swift-nio/pull/240#issuecomment-376703417).